webapp/tests/forms/flows/test_eligibility_step_chooser.py

```diff
@@ -75,3 +76,70 @@ def test_set_attributes_correctly(self):
             self.assertEqual(self.testing_steps, list(step_chooser.steps.values()))
             self.assertEqual(None, step_chooser.overview_step)
 
+
+class TestEligibilityStepChooserDeterminePrevStep(unittest.TestCase):
+    def setUp(self):
+        self.step_chooser = EligibilityStepChooser('eligibility')
+        self.step_chooser.step_order = []
+        self.step_chooser.steps = {}
+
+    def set_up_steps(self, steps_definitions):
+        for step_definition in steps_definitions:
+            step_name = step_definition['name']
+            mocked_step = MagicMock(is_previous_step=MagicMock(return_value=step_definition['is_previous_step']))
+            mocked_step.name = step_name
+            self.step_chooser.step_order.append(step_name)
+            self.step_chooser.steps[step_name] = mocked_step
+
+    def test_if_step_is_first_step_then_return_none(self):
+        self.set_up_steps([
+            {'name': 'step-0', 'is_previous_step': False},
+            {'name': 'step-1', 'is_previous_step': False},
+            {'name': 'step-2', 'is_previous_step': False}
+        ])
+
+        with patch('app.forms.flows.eligibility_step_chooser.get_session_data'):
+            prev_step = self.step_chooser.determine_prev_step('step-0')
+
+        self.assertIsNone(prev_step)
+
+    def test_if_no_step_returns_is_previous_step_true_then_return_first_in_list(self):
+        self.set_up_steps([
+            {'name': 'step-0', 'is_previous_step': False},
+            {'name': 'step-1', 'is_previous_step': False},
+            {'name': 'step-2', 'is_previous_step': False}
+        ])
+
+        with patch('app.forms.flows.eligibility_step_chooser.get_session_data'):
+            prev_step = self.step_chooser.determine_prev_step('step-2')
+
+        self.assertEqual('step-0', prev_step.name)
+
+    def test_is_previous_step_for_correct_steps_are_called(self):
```

Comment:
I would rename it to `test_is_previous_step_only_called_for_given_step_name` because it took me some time to understand what exactly is happening with the `set_up_steps` method.

Reply:
Actually, it should be `test_is_previous_step_only_called_for_previous_steps` (`_except_first_step` if we're very precise). We call the method for step-2 and check for step-1 to be called.

Reply:
Shouldn't we then have more than three steps and test for the third